ui/warrior/presets.ts

```diff
@@ -0,0 +1,427 @@
+import { Warrior_Rotation_FuryRotation_PrimaryInstant as PrimaryInstant } from '/tbc/core/proto/warrior.js';
+
+import * as Enchants from '/tbc/core/constants/enchants.js';
+import * as Gems from '/tbc/core/proto_utils/gems.js';
+import * as Tooltips from '/tbc/core/constants/tooltips.js';
+
+// Preset options for this spec.
+// Eventually we will import these values for the raid sim too, so its good to
+// keep them in a separate file.
+
+// Default talents. Uses the wowhead calculator format, make the talents on
+// https://tbc.wowhead.com/talent-calc and copy the numbers in the url.
+export const ArmsSlamTalents = {
+	name: 'Arms Slam',
+	data: '32003301352010500221-0550000500521203',
+};
+export const ArmsDWTalents = {
+	name: 'Arms DW',
+	data: '32003301352010500221-0550000520501203',
+};
+export const FuryTalents = {
+	name: 'Fury',
+	data: '32003301302-055000055050120531151',
+};
+
+export const DefaultFuryRotation = WarriorRotation.create({
+	type: RotationType.Fury,
+	fury: FuryRotation.create({
+		primaryInstant: PrimaryInstant.Whirlwind,
+		useBtExec: true,
+		rampageCdTresh: 5,
+	}),
+});
+
+export const DefaultFuryOptions = WarriorOptions.create({
+	startingRage: 0,
+	precastSapphire: false,
+	precastT2: false,
+});
+
+export const DefaultFuryConsumes = Consumes.create({
+	defaultPotion: Potions.HastePotion,
+	flaskOfRelentlessAssault: true,
+	roastedClefthoof: true,
+});
+
+export const DefaultArmsSlamRotation = WarriorRotation.create({
+	type: RotationType.ArmsSlam,
+	armsSlam: ArmsSlamRotation.create({
+		useSlamExec: true,
+		slamLatency: 150,
+		useMsExec: true,
+	}),
+});
+
+export const DefaultArmsSlamOptions = WarriorOptions.create({
+	startingRage: 0,
+	precastSapphire: false,
+	precastT2: false,
+});
+
+export const DefaultArmsSlamConsumes = Consumes.create({
+	defaultPotion: Potions.HastePotion,
+	flaskOfRelentlessAssault: true,
+	roastedClefthoof: true,
+});
+
+export const DefaultArmsDWRotation = WarriorRotation.create({
+	type: RotationType.ArmsSlam,
+	armsDw: ArmsDWRotation.create({
+		useMsExec: false,
+	}),
+});
+
+export const DefaultArmsDWOptions = WarriorOptions.create({
+	startingRage: 0,
+	precastSapphire: false,
+	precastT2: false,
+});
+
+export const DefaultArmsDWConsumes = Consumes.create({
+	defaultPotion: Potions.HastePotion,
+	flaskOfRelentlessAssault: true,
+	roastedClefthoof: true,
+});
+
+export const DefaultGeneralRotation = WarriorRotation.create({
+	type: RotationType.General,
+	general: GeneralRotation.create({
+		useWwExec: true,
+		useHsExec: true,
+		useOverpower: false,
+		overpowerRageThresh: 25,
+		useHamstring: false,
+		hamstringRageThresh: 70,
+	}),
+});
+
+export const P1_FURY_PRESET = {
+	name: 'P1 Fury Preset',
+	tooltip: Tooltips.BASIC_BIS_DISCLAIMER,
+	enableWhen: (player: Player<Spec.SpecWarrior>) => player.getRotation().type == RotationType.Fury,
```

Comment:
I'm not going to read through all the item IDs here, but just want to clarify that you dont NEED to have separate presets for each spec. You can change or remove the enableWhen condition if some or all of the specs use the same presets.

If you actually want separate presets for each spec then this is fine.

Reply:
I do want them to have different presets. These specs have different bis sets in while the differences between them are minor.